tsdb/engine/tsm1/engine.go

```diff
@@ -203,6 +204,69 @@ func (e *DevEngine) LoadMetadataIndex(_ *tsdb.Shard, index *tsdb.DatabaseIndex,
 	return nil
 }
 
+// Backup will write a tar archive of any TSM files modified since the passed
+// in time to the passed in writer. The basePath will be prepended to the names
+// of the files in the archive. It will force a snapshot of the WAL first
+// then perform the backup with a read lock against the file store. This means
+// that new TSM files will not be able to be created in this shard while the
+// backup is running. For shards that are still acively getting writes, this
+// could cause the WAL to backup, increasing memory usage and evenutally rejecting writes.
+func (e *DevEngine) Backup(w io.Writer, basePath string, since time.Time) error {
```

Comment:
Is the idea for the Backup function to be a config option, or something that an admin kicks off as necessary? Or will this be an automatic function?

If the latter, perhaps this can be restricted only to shards that are NOT the current shard (ie. duration is passed, kick off backup, continue writing to new shard)? If a single-shard situation, then this should be run only on the very oldest set of tsm files that _should not_ be getting compacted regularly.